src/lib.rs

```diff
@@ -183,33 +172,46 @@ fn run_rustpython(vm: &VirtualMachine, run_mode: RunMode, quiet: bool) -> PyResu
         );
     }
 
-    match run_mode {
+    let is_repl = matches!(run_mode, RunMode::Repl);
+    if !vm.state.settings.quiet
+        && (vm.state.settings.verbose > 0 || (is_repl && std::io::stdin().is_terminal()))
+    {
+        eprintln!(
+            "Welcome to the magnificent Rust Python {} interpreter \u{1f631} \u{1f596}",
+            env!("CARGO_PKG_VERSION")
+        );
+        eprintln!(
+            "RustPython {}.{}.{}",
+            vm::version::MAJOR,
+            vm::version::MINOR,
+            vm::version::MICRO,
+        );
+
+        eprintln!("Type \"help\", \"copyright\", \"credits\" or \"license\" for more information.");
+    }
+    let res = match run_mode {
         RunMode::Command(command) => {
             debug!("Running command {}", command);
-            vm.run_code_string(scope, &command, "<stdin>".to_owned())?;
+            vm.run_code_string(scope.clone(), &command, "<stdin>".to_owned())
+                .map(drop)
```

Comment:
Is this a common convention to drop value from Result? It looks like `drop` is important logic here to me. If not, a weak suggestion.

```suggestion
                .map(|_| ())
```

Reply:
I've seen `.map(drop)` a decent amount, yeah.